Wikipedia/Code/ExploreFeedSettingsViewController.swift

```diff
@@ -0,0 +1,199 @@
+import UIKit
+
+private struct FeedCard: ExploreFeedSettingsSwitchItem {
+    let title: String
+    let subtitle: String?
+    let disclosureType: WMFSettingsMenuItemDisclosureType
+    let disclosureText: String?
+    let type: ExploreFeedSettingsItemType
+    let iconName: String?
+    let iconColor: UIColor?
+    let iconBackgroundColor: UIColor?
+    var controlTag: Int = 0
+    var isOn: Bool = true
+
+    init(contentGroupKind: WMFContentGroupKind, displayType: DisplayType) {
+        type = ExploreFeedSettingsItemType.feedCard(contentGroupKind)
+
+        let languageCodes = SessionSingleton.sharedInstance().dataStore.feedContentController.languageCodes(for: contentGroupKind)
+
+        let disclosureTextString: () -> String = {
+            let preferredLanguages = MWKLanguageLinkController.sharedInstance().preferredLanguages
+            if (languageCodes.count == preferredLanguages.count) {
+                return WMFLocalizedString("explore-feed-preferences-feed-card-visibility-all-languages", value: "On all", comment: "Text for Explore feed card setting indicating that the feed card is active in all preferred languages")
+            } else if languageCodes.count > 0 {
+                return String.localizedStringWithFormat(WMFLocalizedString("explore-feed-preferences-feed-card-visibility-languages-count", value:"On %1$d", comment: "Text for Explore feed card setting indicating the number of languages it's visible in - %1$d is replaced with the number of languages"), languageCodes.count)
+            } else {
+                return WMFLocalizedString("explore-feed-preferences-feed-card-visibility-no-languages", value: "On all", comment: "Text for Explore feed card setting indicating that the feed card is hidden in all preferred languages")
+            }
+        }
+
+        var feedCardDescription: String?
+
+        switch contentGroupKind {
+        case .news:
+            title = CommonStrings.inTheNewsTitle
+            feedCardDescription = WMFLocalizedString("explore-feed-preferences-in-the-news-description", value: "Articles about current events", comment: "Description of In the news section of Explore feed")
+            iconName = "in-the-news-mini"
+            iconColor = UIColor.wmf_lightGray
```

Comment:
If you want to leave out `UIColor` in these case statements I think the compiler will infer it.

Reply:
it's a static let on `UIColor` so it will complain about `Use of unresolved identifier` without `UIColor` 😩 it's easy to take it for an enum though 😄

Reply:
Oh weird! Are you on Xcode 9.4.1? I just tried removing all the switch `UIColor`s and it compiled without warnings...

Reply:
I'm on 9.4, let me check 9.4.1 🧐

Reply:
You were right, it can infer those now 😱👍